sdk/eventhub/Azure.Messaging.EventHubs/src/Azure.Messaging.EventHubs.csproj

```diff
@@ -38,6 +38,7 @@
     <Compile Include="$(AzureCoreSharedSources)AzureEventSource.cs" LinkBase="SharedSource\Azure.Core" />
     <Compile Include="$(AzureCoreSharedSources)HashCodeBuilder.cs" LinkBase="SharedSource\Azure.Core" />
     <Compile Include="$(AzureCoreSharedSources)TaskExtensions.cs" LinkBase="SharedSource\Azure.Core" />
+    <Compile Include="$(AzureCoreSharedSources)TrimmingAttribute.cs" LinkBase="SharedSource\Azure.Core" />
```

Comment:
How do we know when to add this shared source? I don't see how it is used.

Reply:
We need it here because `MessagingClientDiagnostics` uses `DiagnosticScope` which uses `TrimmingAttributes`, it's not super obvious in some cases which file is causing it to be needed, but the build will break with errors saying that the attributes in `DiagnosticScope` are not accessible / could not be found if it needs to be added in.

Reply:
There's a connected component of shared source files around DiagnosticScope.  It's good to know this is one of them, and we can look at reconciling this as part of https://github.com/Azure/azure-sdk-for-net/issues/28369